platform-tests/src/platform/performance/load_test.go

```diff
@@ -35,7 +35,7 @@ func loadTest(appName string, rate uint64, duration time.Duration, keepalive boo
 	for res := range attacker.Attack(targeter, rate, duration) {
 		metrics.Add(res)
 	}
-	return &metrics, metrics.Latencies.P99
```

Comment:
I've figured out why these never failed previously. `metrics.Latencies` is only [populated when `metrics.Close()` is called](https://github.com/tsenart/vegeta/blob/master/lib/metrics.go#L114). Because this is called using `defer` here, it actually runs after the return value is set meaning that this function would always return 0 for the latency.

The new implementation doesn't have this issue.

Reply:
Ah ha, thanks! You're credited in the amended commit message.